app/inbound/process-submit-payment-request.js

```diff
@@ -0,0 +1,27 @@
+const db = require('../data')
+
+const getPaymentRequestByReferenceId = require('./get-payment-request-by-reference-id')
+const saveInvoiceNumber = require('./save-invoice-number')
+const savePaymentRequest = require('./save-payment-request')
+const saveInvoiceLines = require('./save-invoice-lines')
+
+const processSubmitPaymentRequest = async (paymentRequest) => {
+  const transaction = await db.sequelize.transaction()
+  try {
+    const existingPaymentRequest = await getPaymentRequestByReferenceId(paymentRequest.referenceId, transaction)

```

Comment:
I think this is still an issue because the `reference` number will mostly be the same for both the processing and submit messages so whichever got there first would prevent the other being saved.

It's only changed if we need to reprocess a payment request from processing again.  ie DAX has rejected it.

What if we included the status of the payment request in the `paymentRequest` table and used that in the duplicate check?

eg rather than "if this reference exists, it's a duplicate" we have "if this reference exists and it's a submitted payment request" it's a duplicate?